frontend/src/routes/[[lang=locale]]/candidate/(protected)/questions/+page.svelte

```diff
@@ -1,38 +1,217 @@
+
+      loading = true;
+
+      if (answers) {
+        nofUnansweredQuestions = Object.entries(questions).length - Object.entries(answers).length;
+        loading = false;
+        unansweredCategories = Object.keys(questionsByCategory).filter(
+          (category) => !questionsByCategory[category].every((question) => answers?.[question.id])
+        );
+      }
+    }
+  }
 
   const firstQuestionUrl = $getRoute({
     route: Route.CandAppQuestions,
     id: $page.data.questions[0].id
   });
-  const questionCategories = new Set<string>();
-
-  $page.data.questions.forEach((question) => {
-    if (question.category) questionCategories.add(question.category);
-  });
 
   const numQuestions = $page.data.questions.length;
+
+  function getAnsweredButtonText() {
+    if (dataEditable) {
+      return {
+        text: $t('candidateApp.questions.editYourAnswer'),
+        icon: 'missingIcon'
+      };
+    } else {
+      return {
+        text: $t('candidateApp.questions.viewYourAnswer'),
+        icon: 'show'
+      };
+    }
+  }
 </script>
 
-<BasicPage title={$t('candidateApp.opinions.title')}>
-  <svelte:fragment slot="note">
-    <Icon name="tip" />
-    {$t('candidateApp.opinions.tip')}
-  </svelte:fragment>
-
-  <p class="text-center">
-    {$t('candidateApp.opinions.instructions', {numQuestions})}
-  </p>
-
-  <Button
-    slot="primaryActions"
-    href={firstQuestionUrl}
-    variant="main"
-    icon="next"
-    text={$t('candidateApp.opinions.continue')} />
-</BasicPage>
+{#if answers && Object.entries(answers).length === 0}
+  <BasicPage title={$t('candidateApp.opinions.title')}>
+    <svelte:fragment slot="note">
+      <Icon name="tip" />
+      {$t('candidateApp.opinions.tip')}
+    </svelte:fragment>
+    <p class="text-center">
+      {$t('candidateApp.opinions.instructions', {numQuestions})}
+    </p>
+
+    <Button
+      slot="primaryActions"
+      href={firstQuestionUrl}
+      variant="main"
+      icon="next"
+      text={$t('candidateApp.opinions.continue')} />
+  </BasicPage>
+{:else}
+  <BasicPage title={$t('candidateApp.questions.title')}>
+    <Warning display={!dataEditable} slot="note"
+      >{$t('candidateApp.questions.editingAllowedNote')}
+    </Warning>
+
+    <p class="pb-20 text-center">
+      {$t('candidateApp.questions.info')}
+    </p>
+    {#if nofUnansweredQuestions != 0 && !loading && dataEditable}
+      <div class="pb-6 text-center text-warning">
+        {$t('candidateApp.questions.warning', {numUnansweredQuestions: nofUnansweredQuestions})}
+      </div>
+      <div class="flex w-full justify-center pb-40 pt-20">
+        <Button
+          href={$getRoute({route: Route.CandAppQuestions, id: '1'})}
+          text={$t('candidateApp.questions.enterMissingAnswer')}
+          variant="main"
+          icon="next" />
+      </div>
+    {/if}
+
+    {#each Object.entries(questionsByCategory) as [category, categoryQuestions]}
+      <div class="edgetoedge-x">
```

Comment:
You could just use `class='px-0'`

Reply:
'edgetoedge' is a class that overrides padding from higher up and using only 'px-0' does not so it, so 'edgetoedge' is needed here.